python/tester/upload.py

```diff
@@ -39,38 +60,86 @@ class Instrument:
 _log.setLevel(logging.DEBUG)
 
 
-def process_group(publisher, bucket, instrument, group, filter, kind):
-    n_snaps = INSTRUMENTS[instrument].n_snaps
-    send_next_visit(publisher, instrument, group, n_snaps, filter, kind)
+def process_group(publisher, visit_infos, uploader):
+    """Simulate the observation of a single on-sky pointing.
+
+    Parameters
+    ----------
+    publisher : `google.cloud.pubsub_v1.PublisherClient`
+        The client that posts ``next_visit`` messages.
+    visit_infos : `set` [`activator.Visit`]
+        The visit-detector combinations to be observed; each object may
+        represent multiple snaps. Assumed to represent a single group, and to
+        share instrument, snaps, filter, and kind.
+    uploader : callable [`activator.Visit`, int]
+        A callable that takes an exposure spec and a snap ID, and uploads the
+        visit's data.
+    """
+    # Assume group/snaps is shared among all visit_infos
+    for info in visit_infos:
+        group = info.group
+        n_snaps = info.snaps
+        break
+    else:
+        _log.info("No observations to make; aborting.")
+        return
+
+    send_next_visit(publisher, group, visit_infos)
```

Comment:
It's a bit aggressive to be immediately "taking an image" after sending next_visit.  In theory we're supposed to have 20 seconds prior to the exposure being taken, so we could add a 20 sec sleep here.

Reply:
Oh, except that unrealistically gives a 20 second extra delay between groups.  We'd have to do something more complicated with asynchronous or threaded routines.

Reply:
A preexisting bug, but I'm happy to add fix it on this ticket.

Reply:
Agreed on Slack to leave the behavior as-is.